// igniteui-angular/esm2020/lib/grids/grid/grid.details.pipe.mjs
import { Pipe, Inject } from '@angular/core';
import { IGX_GRID_BASE } from '../common/grid.interface';
import * as i0 from "@angular/core";
/** @hidden */
export class IgxGridDetailsPipe {
    constructor(grid) {
        this.grid = grid;
    }
    transform(collection, hasDetails, expansionStates, _pipeTrigger) {
        if (!hasDetails) {
            return collection;
        }
        const res = this.addDetailRows(collection, expansionStates);
        return res;
    }
    addDetailRows(collection, _expansionStates) {
        const result = [];
        collection.forEach((v) => {
            result.push(v);
            if (!this.grid.isGroupByRecord(v) && !this.grid.isSummaryRow(v) &&
                this.grid.gridAPI.get_row_expansion_state(v)) {
                const detailsObj = { detailsData: v };
                result.push(detailsObj);
            }
        });
        return result;
    }
}
IgxGridDetailsPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridDetailsPipe, deps: [{ token: IGX_GRID_BASE }], target: i0.ɵɵFactoryTarget.Pipe });
IgxGridDetailsPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridDetailsPipe, name: "gridDetails" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridDetailsPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'gridDetails' }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [IGX_GRID_BASE]
                }] }]; } });
