// igniteui-angular/esm2020/lib/grids/resizing/resize.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IgxColumnResizingService } from './resizing.service';
import { IgxGridColumnResizerComponent } from './resizer.component';
import { IgxResizeHandleDirective } from './resize-handle.directive';
import { IgxColumnResizerDirective } from './resizer.directive';
import { IgxPivotColumnResizingService } from './pivot-grid/pivot-resizing.service';
import { IgxPivotResizeHandleDirective } from './pivot-grid/pivot-resize-handle.directive';
import { IgxPivotGridColumnResizerComponent } from './pivot-grid/pivot-resizer.component';
import * as i0 from "@angular/core";
export { IgxGridColumnResizerComponent } from './resizer.component';
export { IgxPivotGridColumnResizerComponent } from './pivot-grid/pivot-resizer.component';
export { IgxResizeHandleDirective } from './resize-handle.directive';
export { IgxPivotResizeHandleDirective } from './pivot-grid/pivot-resize-handle.directive';
export { IgxColumnResizerDirective } from './resizer.directive';
export class IgxGridResizingModule {
}
IgxGridResizingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridResizingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxGridResizingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridResizingModule, declarations: [IgxGridColumnResizerComponent,
        IgxResizeHandleDirective,
        IgxColumnResizerDirective,
        IgxPivotGridColumnResizerComponent,
        IgxPivotResizeHandleDirective], imports: [CommonModule], exports: [IgxGridColumnResizerComponent,
        IgxResizeHandleDirective,
        IgxColumnResizerDirective,
        IgxPivotGridColumnResizerComponent,
        IgxPivotResizeHandleDirective] });
IgxGridResizingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridResizingModule, providers: [
        IgxColumnResizingService,
        IgxPivotColumnResizingService
    ], imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridResizingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxGridColumnResizerComponent,
                        IgxResizeHandleDirective,
                        IgxColumnResizerDirective,
                        IgxPivotGridColumnResizerComponent,
                        IgxPivotResizeHandleDirective
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        IgxGridColumnResizerComponent,
                        IgxResizeHandleDirective,
                        IgxColumnResizerDirective,
                        IgxPivotGridColumnResizerComponent,
                        IgxPivotResizeHandleDirective
                    ],
                    providers: [
                        IgxColumnResizingService,
                        IgxPivotColumnResizingService
                    ]
                }]
        }] });
