// igniteui-angular/esm2020/lib/grids/toolbar/toolbar.module.mjs
import { NgModule } from '@angular/core';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnActionsModule } from '../column-actions/column-actions.module';
import { IgxGridToolbarComponent } from './grid-toolbar.component';
import { IgxGridToolbarAdvancedFilteringComponent } from './grid-toolbar-advanced-filtering.component';
import { IgxGridToolbarExporterComponent } from './grid-toolbar-exporter.component';
import { IgxGridToolbarHidingComponent } from './grid-toolbar-hiding.component';
import { IgxGridToolbarPinningComponent } from './grid-toolbar-pinning.component';
import { IgxCSVTextDirective, IgxExcelTextDirective, IgxGridToolbarActionsDirective, IgxGridToolbarDirective, IgxGridToolbarTitleDirective } from './common';
import * as i0 from "@angular/core";
export * from './grid-toolbar.component';
export * from './common';
export * from './grid-toolbar-advanced-filtering.component';
export * from './grid-toolbar-exporter.component';
export * from './grid-toolbar-hiding.component';
export * from './grid-toolbar-pinning.component';
export class IgxGridToolbarModule {
}
IgxGridToolbarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridToolbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxGridToolbarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridToolbarModule, declarations: [IgxCSVTextDirective,
        IgxExcelTextDirective,
        IgxGridToolbarActionsDirective,
        IgxGridToolbarAdvancedFilteringComponent,
        IgxGridToolbarComponent,
        IgxGridToolbarExporterComponent,
        IgxGridToolbarHidingComponent,
        IgxGridToolbarPinningComponent,
        IgxGridToolbarTitleDirective,
        IgxGridToolbarDirective], imports: [IgxColumnActionsModule,
        IgxGridSharedModules], exports: [IgxCSVTextDirective,
        IgxExcelTextDirective,
        IgxGridToolbarActionsDirective,
        IgxGridToolbarAdvancedFilteringComponent,
        IgxGridToolbarComponent,
        IgxGridToolbarExporterComponent,
        IgxGridToolbarHidingComponent,
        IgxGridToolbarPinningComponent,
        IgxGridToolbarTitleDirective,
        IgxGridToolbarDirective] });
IgxGridToolbarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridToolbarModule, imports: [[
            IgxColumnActionsModule,
            IgxGridSharedModules
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridToolbarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxCSVTextDirective,
                        IgxExcelTextDirective,
                        IgxGridToolbarActionsDirective,
                        IgxGridToolbarAdvancedFilteringComponent,
                        IgxGridToolbarComponent,
                        IgxGridToolbarExporterComponent,
                        IgxGridToolbarHidingComponent,
                        IgxGridToolbarPinningComponent,
                        IgxGridToolbarTitleDirective,
                        IgxGridToolbarDirective
                    ],
                    imports: [
                        IgxColumnActionsModule,
                        IgxGridSharedModules
                    ],
                    exports: [
                        IgxCSVTextDirective,
                        IgxExcelTextDirective,
                        IgxGridToolbarActionsDirective,
                        IgxGridToolbarAdvancedFilteringComponent,
                        IgxGridToolbarComponent,
                        IgxGridToolbarExporterComponent,
                        IgxGridToolbarHidingComponent,
                        IgxGridToolbarPinningComponent,
                        IgxGridToolbarTitleDirective,
                        IgxGridToolbarDirective
                    ]
                }]
        }] });
