// igniteui-angular/esm2020/lib/carousel/carousel.directives.mjs
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
export class IgxCarouselIndicatorDirective {
}
IgxCarouselIndicatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxCarouselIndicatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
IgxCarouselIndicatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxCarouselIndicatorDirective, selector: "[igxCarouselIndicator]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxCarouselIndicatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[igxCarouselIndicator]'
                }]
        }] });
export class IgxCarouselNextButtonDirective {
}
IgxCarouselNextButtonDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxCarouselNextButtonDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
IgxCarouselNextButtonDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxCarouselNextButtonDirective, selector: "[igxCarouselNextButton]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxCarouselNextButtonDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[igxCarouselNextButton]'
                }]
        }] });
export class IgxCarouselPrevButtonDirective {
}
IgxCarouselPrevButtonDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxCarouselPrevButtonDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
IgxCarouselPrevButtonDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxCarouselPrevButtonDirective, selector: "[igxCarouselPrevButton]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxCarouselPrevButtonDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[igxCarouselPrevButton]'
                }]
        }] });
