// igniteui-angular/esm2020/lib/splitter/splitter.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IgxSplitterPaneComponent } from './splitter-pane/splitter-pane.component';
import { IgxSplitterComponent, IgxSplitBarComponent } from './splitter.component';
import { IgxIconModule } from '../icon/public_api';
import { IgxDragDropModule } from '../directives/drag-drop/drag-drop.directive';
import * as i0 from "@angular/core";
export class IgxSplitterModule {
}
IgxSplitterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxSplitterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxSplitterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxSplitterModule, declarations: [IgxSplitterComponent,
        IgxSplitterPaneComponent,
        IgxSplitBarComponent], imports: [CommonModule, IgxIconModule, IgxDragDropModule], exports: [IgxSplitterComponent,
        IgxSplitterPaneComponent] });
IgxSplitterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxSplitterModule, imports: [[
            CommonModule, IgxIconModule, IgxDragDropModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxSplitterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule, IgxIconModule, IgxDragDropModule
                    ],
                    declarations: [
                        IgxSplitterComponent,
                        IgxSplitterPaneComponent,
                        IgxSplitBarComponent
                    ],
                    exports: [
                        IgxSplitterComponent,
                        IgxSplitterPaneComponent
                    ]
                }]
        }] });
