// igniteui-angular/esm2020/lib/accordion/accordion.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IgxExpansionPanelModule } from '../expansion-panel/expansion-panel.module';
import { IgxAccordionComponent } from './accordion.component';
import * as i0 from "@angular/core";
/**
 * @hidden
 */
export class IgxAccordionModule {
}
IgxAccordionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxAccordionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxAccordionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxAccordionModule, declarations: [IgxAccordionComponent], imports: [IgxExpansionPanelModule,
        CommonModule], exports: [IgxAccordionComponent,
        IgxExpansionPanelModule] });
IgxAccordionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxAccordionModule, imports: [[
            IgxExpansionPanelModule,
            CommonModule,
        ], IgxExpansionPanelModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxAccordionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxAccordionComponent
                    ],
                    imports: [
                        IgxExpansionPanelModule,
                        CommonModule,
                    ],
                    exports: [
                        IgxAccordionComponent,
                        IgxExpansionPanelModule
                    ]
                }]
        }] });
