// igniteui-angular/esm2020/lib/progressbar/progressbar.common.mjs
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
export class IgxProcessBarTextTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxProcessBarTextTemplateDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxProcessBarTextTemplateDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
IgxProcessBarTextTemplateDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxProcessBarTextTemplateDirective, selector: "[igxProcessBarText]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxProcessBarTextTemplateDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[igxProcessBarText]'
                }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
export class IgxProgressBarGradientDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxProgressBarGradientDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxProgressBarGradientDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
IgxProgressBarGradientDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxProgressBarGradientDirective, selector: "[igxProgressBarGradient]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxProgressBarGradientDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[igxProgressBarGradient]'
                }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
