// igniteui-angular/esm2020/lib/tabs/tab-header.directive.mjs
import { Directive, HostBinding, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./tabs.base";
import * as i2 from "./tab-item.directive";
import * as i3 from "../core/utils";
export class IgxTabHeaderDirective {
    /** @hidden */
    constructor(tabs, tab, elementRef, platform) {
        this.tabs = tabs;
        this.tab = tab;
        this.elementRef = elementRef;
        this.platform = platform;
        /** @hidden */
        this.role = 'tab';
    }
    /** @hidden */
    get tabIndex() {
        return this.tab.selected ? 0 : -1;
    }
    /** @hidden */
    get ariaSelected() {
        return this.tab.selected;
    }
    /** @hidden */
    get ariaDisabled() {
        return this.tab.disabled;
    }
    /** @hidden */
    onClick() {
        if (this.tab.panelComponent) {
            this.tabs.selectTab(this.tab, true);
        }
    }
    /** @hidden */
    get nativeElement() {
        return this.elementRef.nativeElement;
    }
    ;
}
IgxTabHeaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabHeaderDirective, deps: [{ token: i1.IgxTabsBase }, { token: i2.IgxTabItemDirective }, { token: i0.ElementRef }, { token: i3.PlatformUtil }], target: i0.ɵɵFactoryTarget.Directive });
IgxTabHeaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxTabHeaderDirective, host: { listeners: { "click": "onClick()" }, properties: { "attr.role": "this.role", "attr.tabindex": "this.tabIndex", "attr.aria-selected": "this.ariaSelected", "attr.aria-disabled": "this.ariaDisabled" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabHeaderDirective, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.IgxTabsBase }, { type: i2.IgxTabItemDirective }, { type: i0.ElementRef }, { type: i3.PlatformUtil }]; }, propDecorators: { role: [{
                type: HostBinding,
                args: ['attr.role']
            }], tabIndex: [{
                type: HostBinding,
                args: ['attr.tabindex']
            }], ariaSelected: [{
                type: HostBinding,
                args: ['attr.aria-selected']
            }], ariaDisabled: [{
                type: HostBinding,
                args: ['attr.aria-disabled']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
