// igniteui-angular/esm2020/lib/tabs/tab-content.directive.mjs
import { Directive, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./tab-item.directive";
export class IgxTabContentDirective {
    /** @hidden */
    constructor(tab, elementRef) {
        this.tab = tab;
        this.elementRef = elementRef;
        /** @hidden */
        this.role = 'tabpanel';
    }
    /** @hidden */
    get tabIndex() {
        return this.tab.selected ? 0 : -1;
    }
    /** @hidden */
    get zIndex() {
        return this.tab.selected ? 'auto' : -1;
    }
    /** @hidden */
    get nativeElement() {
        return this.elementRef.nativeElement;
    }
    ;
}
IgxTabContentDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabContentDirective, deps: [{ token: i1.IgxTabItemDirective }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
IgxTabContentDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxTabContentDirective, host: { properties: { "attr.role": "this.role", "attr.tabindex": "this.tabIndex", "style.z-index": "this.zIndex" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabContentDirective, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.IgxTabItemDirective }, { type: i0.ElementRef }]; }, propDecorators: { role: [{
                type: HostBinding,
                args: ['attr.role']
            }], tabIndex: [{
                type: HostBinding,
                args: ['attr.tabindex']
            }], zIndex: [{
                type: HostBinding,
                args: ['style.z-index']
            }] } });
