// igniteui-angular/esm2020/lib/grids/headers/headers.module.mjs
import { NgModule } from '@angular/core';
import { IgxGridHeaderComponent } from './grid-header.component';
import { IgxGridHeaderGroupComponent } from './grid-header-group.component';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnMovingModule } from '../moving/moving.module';
import { IgxGridFilteringModule } from '../filtering/base/filtering.module';
import { IgxGridResizingModule } from '../resizing/resize.module';
import { IgxHeaderGroupStylePipe, IgxHeaderGroupWidthPipe, SortingIndexPipe } from './pipes';
import { IgxGridPipesModule } from '../common/grid-pipes.module';
import { IgxGridHeaderRowComponent } from './grid-header-row.component';
import * as i0 from "@angular/core";
export * from './grid-header-group.component';
export * from './grid-header.component';
export * from './pipes';
export { IgxGridHeaderComponent } from './grid-header.component';
export { IgxGridHeaderGroupComponent } from './grid-header-group.component';
export { IgxGridHeaderRowComponent } from './grid-header-row.component';
export class IgxGridHeadersModule {
}
IgxGridHeadersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridHeadersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxGridHeadersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridHeadersModule, declarations: [IgxGridHeaderComponent,
        IgxGridHeaderGroupComponent,
        IgxGridHeaderRowComponent,
        SortingIndexPipe,
        IgxHeaderGroupWidthPipe,
        IgxHeaderGroupStylePipe], imports: [IgxGridSharedModules,
        IgxGridFilteringModule,
        IgxColumnMovingModule,
        IgxGridResizingModule,
        IgxGridPipesModule], exports: [IgxGridHeaderComponent,
        IgxGridHeaderGroupComponent,
        IgxGridHeaderRowComponent,
        IgxHeaderGroupWidthPipe,
        SortingIndexPipe,
        IgxHeaderGroupStylePipe] });
IgxGridHeadersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridHeadersModule, imports: [[
            IgxGridSharedModules,
            IgxGridFilteringModule,
            IgxColumnMovingModule,
            IgxGridResizingModule,
            IgxGridPipesModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridHeadersModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxGridHeaderComponent,
                        IgxGridHeaderGroupComponent,
                        IgxGridHeaderRowComponent,
                        SortingIndexPipe,
                        IgxHeaderGroupWidthPipe,
                        IgxHeaderGroupStylePipe
                    ],
                    imports: [
                        IgxGridSharedModules,
                        IgxGridFilteringModule,
                        IgxColumnMovingModule,
                        IgxGridResizingModule,
                        IgxGridPipesModule
                    ],
                    exports: [
                        IgxGridHeaderComponent,
                        IgxGridHeaderGroupComponent,
                        IgxGridHeaderRowComponent,
                        IgxHeaderGroupWidthPipe,
                        SortingIndexPipe,
                        IgxHeaderGroupStylePipe
                    ]
                }]
        }] });
