// igniteui-angular/esm2020/lib/tabs/tabs/tabs.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IgxRippleModule } from '../../directives/ripple/ripple.directive';
import { IgxIconModule } from '../../icon/public_api';
import { IgxTabHeaderComponent } from './tab-header.component';
import { IgxTabHeaderIconDirective, IgxTabHeaderLabelDirective } from './tabs.directives';
import { IgxTabItemComponent } from './tab-item.component';
import { IgxTabContentComponent } from './tab-content.component';
import { IgxTabsComponent } from './tabs.component';
import { IgxPrefixModule } from '../../directives/prefix/prefix.directive';
import { IgxSuffixModule } from '../../directives/suffix/suffix.directive';
import * as i0 from "@angular/core";
/** @hidden */
export class IgxTabsModule {
}
IgxTabsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxTabsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabsModule, declarations: [IgxTabsComponent,
        IgxTabItemComponent,
        IgxTabHeaderComponent,
        IgxTabContentComponent,
        IgxTabHeaderLabelDirective,
        IgxTabHeaderIconDirective], imports: [CommonModule, IgxIconModule, IgxRippleModule, IgxPrefixModule, IgxSuffixModule], exports: [IgxTabsComponent,
        IgxTabItemComponent,
        IgxTabHeaderComponent,
        IgxTabContentComponent,
        IgxTabHeaderLabelDirective,
        IgxTabHeaderIconDirective,
        IgxPrefixModule,
        IgxSuffixModule] });
IgxTabsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabsModule, imports: [[CommonModule, IgxIconModule, IgxRippleModule, IgxPrefixModule, IgxSuffixModule], IgxPrefixModule,
        IgxSuffixModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxTabsComponent,
                        IgxTabItemComponent,
                        IgxTabHeaderComponent,
                        IgxTabContentComponent,
                        IgxTabHeaderLabelDirective,
                        IgxTabHeaderIconDirective
                    ],
                    exports: [
                        IgxTabsComponent,
                        IgxTabItemComponent,
                        IgxTabHeaderComponent,
                        IgxTabContentComponent,
                        IgxTabHeaderLabelDirective,
                        IgxTabHeaderIconDirective,
                        IgxPrefixModule,
                        IgxSuffixModule
                    ],
                    imports: [CommonModule, IgxIconModule, IgxRippleModule, IgxPrefixModule, IgxSuffixModule]
                }]
        }] });
