// igniteui-angular/esm2020/lib/date-picker/date-picker.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { IgxCalendarModule } from '../calendar/public_api';
import { IgxCalendarContainerModule } from '../date-common/calendar-container/calendar-container.component';
import { IgxPickersCommonModule } from '../date-common/picker-icons.common';
import { IgxDateTimeEditorModule } from '../directives/date-time-editor/public_api';
import { IgxMaskModule } from '../directives/mask/mask.directive';
import { IgxTextSelectionModule } from '../directives/text-selection/text-selection.directive';
import { IgxIconModule } from '../icon/public_api';
import { IgxInputGroupModule } from '../input-group/public_api';
import { IgxDatePickerComponent } from './date-picker.component';
import * as i0 from "@angular/core";
/** @hidden */
export class IgxDatePickerModule {
}
IgxDatePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDatePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxDatePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDatePickerModule, declarations: [IgxDatePickerComponent], imports: [FormsModule,
        CommonModule,
        IgxIconModule,
        IgxMaskModule,
        IgxCalendarModule,
        IgxInputGroupModule,
        IgxPickersCommonModule,
        IgxTextSelectionModule,
        IgxDateTimeEditorModule,
        IgxCalendarContainerModule], exports: [IgxDatePickerComponent,
        IgxPickersCommonModule] });
IgxDatePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDatePickerModule, imports: [[
            FormsModule,
            CommonModule,
            IgxIconModule,
            IgxMaskModule,
            IgxCalendarModule,
            IgxInputGroupModule,
            IgxPickersCommonModule,
            IgxTextSelectionModule,
            IgxDateTimeEditorModule,
            IgxCalendarContainerModule,
        ], IgxPickersCommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDatePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxDatePickerComponent
                    ],
                    exports: [
                        IgxDatePickerComponent,
                        IgxPickersCommonModule
                    ],
                    imports: [
                        FormsModule,
                        CommonModule,
                        IgxIconModule,
                        IgxMaskModule,
                        IgxCalendarModule,
                        IgxInputGroupModule,
                        IgxPickersCommonModule,
                        IgxTextSelectionModule,
                        IgxDateTimeEditorModule,
                        IgxCalendarContainerModule,
                    ]
                }]
        }] });
