// igniteui-angular/esm2020/lib/grids/summaries/summary.module.mjs
import { NgModule } from '@angular/core';
import { IgxSummaryCellComponent } from './summary-cell.component';
import { IgxSummaryRowComponent } from './summary-row.component';
import { IgxSummaryDataPipe } from './grid-root-summary.pipe';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxGridPipesModule } from '../common/grid-pipes.module';
import * as i0 from "@angular/core";
export { IgxSummaryCellComponent } from './summary-cell.component';
export { IgxSummaryRowComponent } from './summary-row.component';
export { IgxSummaryDataPipe } from './grid-root-summary.pipe';
export class IgxGridSummaryModule {
}
IgxGridSummaryModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSummaryModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxGridSummaryModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSummaryModule, declarations: [IgxSummaryCellComponent,
        IgxSummaryRowComponent,
        IgxSummaryDataPipe], imports: [IgxGridPipesModule,
        IgxGridSharedModules], exports: [IgxSummaryCellComponent,
        IgxSummaryRowComponent,
        IgxSummaryDataPipe] });
IgxGridSummaryModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSummaryModule, imports: [[
            IgxGridPipesModule,
            IgxGridSharedModules
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSummaryModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxSummaryCellComponent,
                        IgxSummaryRowComponent,
                        IgxSummaryDataPipe
                    ],
                    imports: [
                        IgxGridPipesModule,
                        IgxGridSharedModules
                    ],
                    exports: [
                        IgxSummaryCellComponent,
                        IgxSummaryRowComponent,
                        IgxSummaryDataPipe
                    ]
                }]
        }] });
