// igniteui-angular/esm2020/lib/date-range-picker/date-range-picker.module.mjs
import { NgModule } from '@angular/core';
import { IgxDateRangePickerComponent } from './date-range-picker.component';
import { IgxCalendarModule } from '../calendar/public_api';
import { IgxToggleModule } from '../directives/toggle/toggle.directive';
import { CommonModule } from '@angular/common';
import { IgxButtonModule } from '../directives/button/button.directive';
import { IgxInputGroupModule } from '../input-group/public_api';
import { IgxIconModule } from '../icon/public_api';
import { IgxDateRangeStartComponent, IgxDateRangeEndComponent, DateRangePickerFormatPipe, IgxDateRangeSeparatorDirective, IgxDateRangeInputsBaseComponent } from './date-range-picker-inputs.common';
import { IgxDateTimeEditorModule } from '../directives/date-time-editor/public_api';
import { IgxPickersCommonModule } from '../date-common/picker-icons.common';
import { IgxCalendarContainerModule } from '../date-common/calendar-container/calendar-container.component';
import * as i0 from "@angular/core";
/** @hidden */
export class IgxDateRangePickerModule {
}
IgxDateRangePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDateRangePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxDateRangePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDateRangePickerModule, declarations: [IgxDateRangePickerComponent,
        IgxDateRangeStartComponent,
        IgxDateRangeEndComponent,
        IgxDateRangeInputsBaseComponent,
        DateRangePickerFormatPipe,
        IgxDateRangeSeparatorDirective], imports: [CommonModule,
        IgxIconModule,
        IgxButtonModule,
        IgxToggleModule,
        IgxCalendarModule,
        IgxInputGroupModule,
        IgxPickersCommonModule,
        IgxDateTimeEditorModule,
        IgxCalendarContainerModule], exports: [IgxDateRangePickerComponent,
        IgxDateRangeStartComponent,
        IgxDateRangeEndComponent,
        IgxDateRangeSeparatorDirective,
        IgxDateTimeEditorModule,
        IgxPickersCommonModule,
        IgxInputGroupModule] });
IgxDateRangePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDateRangePickerModule, imports: [[
            CommonModule,
            IgxIconModule,
            IgxButtonModule,
            IgxToggleModule,
            IgxCalendarModule,
            IgxInputGroupModule,
            IgxPickersCommonModule,
            IgxDateTimeEditorModule,
            IgxCalendarContainerModule
        ], IgxDateTimeEditorModule,
        IgxPickersCommonModule,
        IgxInputGroupModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDateRangePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxDateRangePickerComponent,
                        IgxDateRangeStartComponent,
                        IgxDateRangeEndComponent,
                        IgxDateRangeInputsBaseComponent,
                        DateRangePickerFormatPipe,
                        IgxDateRangeSeparatorDirective,
                    ],
                    imports: [
                        CommonModule,
                        IgxIconModule,
                        IgxButtonModule,
                        IgxToggleModule,
                        IgxCalendarModule,
                        IgxInputGroupModule,
                        IgxPickersCommonModule,
                        IgxDateTimeEditorModule,
                        IgxCalendarContainerModule
                    ],
                    exports: [
                        IgxDateRangePickerComponent,
                        IgxDateRangeStartComponent,
                        IgxDateRangeEndComponent,
                        IgxDateRangeSeparatorDirective,
                        IgxDateTimeEditorModule,
                        IgxPickersCommonModule,
                        IgxInputGroupModule
                    ]
                }]
        }] });
