// igniteui-angular/esm2020/lib/tabs/bottom-nav/bottom-nav-header.component.mjs
import { Component, HostBinding } from '@angular/core';
import { IgxTabHeaderDirective } from '../tab-header.directive';
import { IgxTabHeaderBase } from '../tabs.base';
import * as i0 from "@angular/core";
export class IgxBottomNavHeaderComponent extends IgxTabHeaderDirective {
    /** @hidden */
    get cssClassSelected() {
        return this.tab.selected;
    }
    /** @hidden */
    get cssClassDisabled() {
        return this.tab.disabled;
    }
    /** @hidden */
    get cssClass() {
        return (!this.tab.disabled && !this.tab.selected);
    }
}
IgxBottomNavHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxBottomNavHeaderComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
IgxBottomNavHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.2.2", type: IgxBottomNavHeaderComponent, selector: "igx-bottom-nav-header", host: { properties: { "class.igx-bottom-nav__menu-item--selected": "this.cssClassSelected", "class.igx-bottom-nav__menu-item--disabled": "this.cssClassDisabled", "class.igx-bottom-nav__menu-item": "this.cssClass" } }, providers: [{ provide: IgxTabHeaderBase, useExisting: IgxBottomNavHeaderComponent }], usesInheritance: true, ngImport: i0, template: "<ng-content></ng-content>\n" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxBottomNavHeaderComponent, decorators: [{
            type: Component,
            args: [{ selector: 'igx-bottom-nav-header', providers: [{ provide: IgxTabHeaderBase, useExisting: IgxBottomNavHeaderComponent }], template: "<ng-content></ng-content>\n" }]
        }], propDecorators: { cssClassSelected: [{
                type: HostBinding,
                args: ['class.igx-bottom-nav__menu-item--selected']
            }], cssClassDisabled: [{
                type: HostBinding,
                args: ['class.igx-bottom-nav__menu-item--disabled']
            }], cssClass: [{
                type: HostBinding,
                args: ['class.igx-bottom-nav__menu-item']
            }] } });
