// igniteui-angular/esm2020/lib/tabs/bottom-nav/bottom-nav.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IgxRippleModule } from '../../directives/ripple/ripple.directive';
import { IgxIconModule } from '../../icon/public_api';
import { IgxBottomNavHeaderComponent } from './bottom-nav-header.component';
import { IgxBottomNavHeaderIconDirective, IgxBottomNavHeaderLabelDirective } from './bottom-nav.directives';
import { IgxBottomNavItemComponent } from './bottom-nav-item.component';
import { IgxBottomNavContentComponent } from './bottom-nav-content.component';
import { IgxBottomNavComponent } from './bottom-nav.component';
import * as i0 from "@angular/core";
/** @hidden */
export class IgxBottomNavModule {
}
IgxBottomNavModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxBottomNavModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxBottomNavModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxBottomNavModule, declarations: [IgxBottomNavComponent,
        IgxBottomNavItemComponent,
        IgxBottomNavHeaderComponent,
        IgxBottomNavContentComponent,
        IgxBottomNavHeaderLabelDirective,
        IgxBottomNavHeaderIconDirective], imports: [CommonModule, IgxIconModule, IgxRippleModule], exports: [IgxBottomNavComponent,
        IgxBottomNavItemComponent,
        IgxBottomNavHeaderComponent,
        IgxBottomNavContentComponent,
        IgxBottomNavHeaderLabelDirective,
        IgxBottomNavHeaderIconDirective] });
IgxBottomNavModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxBottomNavModule, imports: [[CommonModule, IgxIconModule, IgxRippleModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxBottomNavModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxBottomNavComponent,
                        IgxBottomNavItemComponent,
                        IgxBottomNavHeaderComponent,
                        IgxBottomNavContentComponent,
                        IgxBottomNavHeaderLabelDirective,
                        IgxBottomNavHeaderIconDirective
                    ],
                    exports: [
                        IgxBottomNavComponent,
                        IgxBottomNavItemComponent,
                        IgxBottomNavHeaderComponent,
                        IgxBottomNavContentComponent,
                        IgxBottomNavHeaderLabelDirective,
                        IgxBottomNavHeaderIconDirective
                    ],
                    imports: [CommonModule, IgxIconModule, IgxRippleModule]
                }]
        }] });
