// igniteui-angular/esm2020/lib/directives/prefix/prefix.directive.mjs
import { NgModule, Directive } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @hidden
 */
export class IgxPrefixDirective {
}
IgxPrefixDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPrefixDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
IgxPrefixDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxPrefixDirective, selector: "igx-prefix,[igxPrefix]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPrefixDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'igx-prefix,[igxPrefix]'
                }]
        }] });
/**
 * @hidden
 */
export class IgxPrefixModule {
}
IgxPrefixModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPrefixModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxPrefixModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPrefixModule, declarations: [IgxPrefixDirective], exports: [IgxPrefixDirective] });
IgxPrefixModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPrefixModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPrefixModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [IgxPrefixDirective],
                    exports: [IgxPrefixDirective]
                }]
        }] });
