// igniteui-angular/esm2020/lib/drop-down/public_api.mjs
import { NgModule } from '@angular/core';
import { IgxDropDownComponent } from './drop-down.component';
import { IgxDropDownItemComponent } from './drop-down-item.component';
import { IgxDropDownItemNavigationDirective } from './drop-down-navigation.directive';
import { CommonModule } from '@angular/common';
import { IgxToggleModule } from '../directives/toggle/toggle.directive';
import { IgxDropDownGroupComponent } from './drop-down-group.component';
import { IgxDropDownItemBaseDirective } from './drop-down-item.base';
import * as i0 from "@angular/core";
export * from './drop-down.component';
export * from './drop-down-item.component';
export * from './drop-down-navigation.directive';
export * from './drop-down.base';
export * from './drop-down-item.base';
export * from './drop-down-group.component';
/**
 * @hidden
 */
export class IgxDropDownModule {
}
IgxDropDownModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDropDownModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxDropDownModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDropDownModule, declarations: [IgxDropDownComponent,
        IgxDropDownItemBaseDirective,
        IgxDropDownItemComponent,
        IgxDropDownGroupComponent,
        IgxDropDownItemNavigationDirective], imports: [CommonModule,
        IgxToggleModule], exports: [IgxDropDownComponent,
        IgxDropDownItemComponent,
        IgxDropDownGroupComponent,
        IgxDropDownItemNavigationDirective] });
IgxDropDownModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDropDownModule, imports: [[
            CommonModule,
            IgxToggleModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxDropDownModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxDropDownComponent,
                        IgxDropDownItemBaseDirective,
                        IgxDropDownItemComponent,
                        IgxDropDownGroupComponent,
                        IgxDropDownItemNavigationDirective
                    ],
                    exports: [
                        IgxDropDownComponent,
                        IgxDropDownItemComponent,
                        IgxDropDownGroupComponent,
                        IgxDropDownItemNavigationDirective
                    ],
                    imports: [
                        CommonModule,
                        IgxToggleModule
                    ]
                }]
        }] });
