// igniteui-angular/esm2020/lib/paginator/public_api.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { IgxSelectModule } from '../select/public_api';
import { IgxIconModule } from '../icon/public_api';
import { IgxButtonModule } from '../directives/button/button.directive';
import { IgxRippleModule } from '../directives/ripple/ripple.directive';
import { IgxInputGroupModule } from '../input-group/public_api';
import { IgxPageNavigationComponent, IgxPageSizeSelectorComponent, IgxPaginatorComponent, IgxPaginatorTemplateDirective } from './paginator.component';
import { IgxPaginatorDirective } from './paginator-interfaces';
import * as i0 from "@angular/core";
export * from './paginator.component';
export class IgxPaginatorModule {
}
IgxPaginatorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPaginatorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxPaginatorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPaginatorModule, declarations: [IgxPaginatorComponent,
        IgxPageNavigationComponent,
        IgxPageSizeSelectorComponent,
        IgxPaginatorTemplateDirective,
        IgxPaginatorDirective], imports: [CommonModule,
        FormsModule,
        IgxButtonModule,
        IgxIconModule,
        IgxInputGroupModule,
        IgxRippleModule,
        IgxSelectModule], exports: [IgxPaginatorComponent,
        IgxPageNavigationComponent,
        IgxPageSizeSelectorComponent,
        IgxPaginatorTemplateDirective,
        IgxPaginatorDirective] });
IgxPaginatorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPaginatorModule, imports: [[
            CommonModule,
            FormsModule,
            IgxButtonModule,
            IgxIconModule,
            IgxInputGroupModule,
            IgxRippleModule,
            IgxSelectModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPaginatorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxPaginatorComponent,
                        IgxPageNavigationComponent,
                        IgxPageSizeSelectorComponent,
                        IgxPaginatorTemplateDirective,
                        IgxPaginatorDirective
                    ],
                    exports: [
                        IgxPaginatorComponent,
                        IgxPageNavigationComponent,
                        IgxPageSizeSelectorComponent,
                        IgxPaginatorTemplateDirective,
                        IgxPaginatorDirective
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        IgxButtonModule,
                        IgxIconModule,
                        IgxInputGroupModule,
                        IgxRippleModule,
                        IgxSelectModule
                    ]
                }]
        }] });
