// igniteui-angular/esm2020/lib/directives/notification/notifications.directive.mjs
import { Directive, HostBinding, Input } from '@angular/core';
import { Subject } from 'rxjs';
import { IgxToggleDirective } from '../toggle/toggle.directive';
import * as i0 from "@angular/core";
export class IgxNotificationsDirective extends IgxToggleDirective {
    constructor() {
        super(...arguments);
        /**
         * Sets/gets the `aria-live` attribute.
         * If not set, `aria-live` will have value `"polite"`.
         */
        this.ariaLive = 'polite';
        /**
         * Sets/gets whether the element will be hidden after the `displayTime` is over.
         * Default value is `true`.
         */
        this.autoHide = true;
        /**
         * Sets/gets the duration of time span (in milliseconds) which the element will be visible
         * after it is being shown.
         * Default value is `4000`.
         */
        this.displayTime = 4000;
        /**
         * @hidden
         * @internal
         */
        this.textMessage = '';
        this.d$ = new Subject();
    }
    /**
     * Enables/Disables the visibility of the element.
     * If not set, the `isVisible` attribute will have value `false`.
     */
    get isVisible() {
        return !this.collapsed;
    }
    set isVisible(value) {
        if (value !== this.isVisible) {
            if (value) {
                requestAnimationFrame(() => {
                    this.open();
                });
            }
            else {
                this.close();
            }
        }
    }
    /**
     * @hidden
     */
    open() {
        clearInterval(this.timeoutId);
        const overlaySettings = {
            positionStrategy: this.strategy,
            closeOnEscape: false,
            closeOnOutsideClick: false,
            modal: false,
            outlet: this.outlet
        };
        super.open(overlaySettings);
        if (this.autoHide) {
            this.timeoutId = window.setTimeout(() => {
                this.close();
            }, this.displayTime);
        }
    }
    /**
     * Hides the element.
     */
    close() {
        clearTimeout(this.timeoutId);
        super.close();
    }
    /**
     * @hidden
     */
    ngOnDestroy() {
        this.d$.next(true);
        this.d$.complete();
    }
}
IgxNotificationsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxNotificationsDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
IgxNotificationsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxNotificationsDirective, inputs: { ariaLive: "ariaLive", autoHide: "autoHide", displayTime: "displayTime", outlet: "outlet", isVisible: "isVisible" }, host: { properties: { "attr.aria-live": "this.ariaLive" } }, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxNotificationsDirective, decorators: [{
            type: Directive
        }], propDecorators: { ariaLive: [{
                type: HostBinding,
                args: ['attr.aria-live']
            }, {
                type: Input
            }], autoHide: [{
                type: Input
            }], displayTime: [{
                type: Input
            }], outlet: [{
                type: Input
            }], isVisible: [{
                type: Input
            }] } });
