// igniteui-angular/esm2020/lib/grids/column-actions/column-actions.module.mjs
import { NgModule } from '@angular/core';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnHidingDirective } from './column-hiding.directive';
import { IgxColumnPinningDirective } from './column-pinning.directive';
import { IgxColumnActionEnabledPipe, IgxColumnActionsComponent, IgxFilterActionColumnsPipe, IgxSortActionColumnsPipe } from './column-actions.component';
import { IgxGridPipesModule } from '../common/grid-pipes.module';
import * as i0 from "@angular/core";
export * from './column-actions.component';
export * from './column-hiding.directive';
export * from './column-pinning.directive';
export class IgxColumnActionsModule {
}
IgxColumnActionsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnActionsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxColumnActionsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnActionsModule, declarations: [IgxColumnHidingDirective,
        IgxColumnPinningDirective,
        IgxColumnActionsComponent,
        IgxColumnActionEnabledPipe,
        IgxFilterActionColumnsPipe,
        IgxSortActionColumnsPipe], imports: [IgxGridSharedModules,
        IgxGridPipesModule], exports: [IgxColumnHidingDirective,
        IgxColumnPinningDirective,
        IgxColumnActionsComponent,
        IgxColumnActionEnabledPipe,
        IgxFilterActionColumnsPipe,
        IgxSortActionColumnsPipe] });
IgxColumnActionsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnActionsModule, imports: [[
            IgxGridSharedModules,
            IgxGridPipesModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnActionsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxColumnHidingDirective,
                        IgxColumnPinningDirective,
                        IgxColumnActionsComponent,
                        IgxColumnActionEnabledPipe,
                        IgxFilterActionColumnsPipe,
                        IgxSortActionColumnsPipe
                    ],
                    imports: [
                        IgxGridSharedModules,
                        IgxGridPipesModule
                    ],
                    exports: [
                        IgxColumnHidingDirective,
                        IgxColumnPinningDirective,
                        IgxColumnActionsComponent,
                        IgxColumnActionEnabledPipe,
                        IgxFilterActionColumnsPipe,
                        IgxSortActionColumnsPipe
                    ]
                }]
        }] });
