// igniteui-angular/esm2020/lib/grids/filtering/base/filtering.module.mjs
import { NgModule } from '@angular/core';
import { IgxGridFilteringCellComponent } from './grid-filtering-cell.component';
import { IgxGridFilteringRowComponent } from './grid-filtering-row.component';
import { IgxGridSharedModules } from '../../common/shared.module';
import { IgxGridPipesModule } from '../../common/grid-pipes.module';
import * as i0 from "@angular/core";
export { IgxGridFilteringCellComponent as θIgxGridFilteringCellComponent } from './grid-filtering-cell.component';
export { IgxGridFilteringRowComponent as θIgxGridFilteringRowComponent } from './grid-filtering-row.component';
export class IgxGridFilteringModule {
}
IgxGridFilteringModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridFilteringModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxGridFilteringModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridFilteringModule, declarations: [IgxGridFilteringCellComponent,
        IgxGridFilteringRowComponent], imports: [IgxGridSharedModules,
        IgxGridPipesModule], exports: [IgxGridFilteringCellComponent,
        IgxGridFilteringRowComponent] });
IgxGridFilteringModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridFilteringModule, imports: [[
            IgxGridSharedModules,
            IgxGridPipesModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridFilteringModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxGridFilteringCellComponent,
                        IgxGridFilteringRowComponent
                    ],
                    imports: [
                        IgxGridSharedModules,
                        IgxGridPipesModule
                    ],
                    exports: [
                        IgxGridFilteringCellComponent,
                        IgxGridFilteringRowComponent
                    ]
                }]
        }] });
