// igniteui-angular/esm2020/lib/grids/resizing/pivot-grid/pivot-resize-handle.directive.mjs
import { Directive, Input } from '@angular/core';
import { IgxResizeHandleDirective } from '../resize-handle.directive';
import * as i0 from "@angular/core";
import * as i1 from "./pivot-resizing.service";
/**
 * @hidden
 * @internal
 */
export class IgxPivotResizeHandleDirective extends IgxResizeHandleDirective {
    constructor(zone, element, colResizingService) {
        super(zone, element, colResizingService);
        this.zone = zone;
        this.element = element;
        this.colResizingService = colResizingService;
    }
    /**
     * @hidden
     */
    set pivotColumn(value) {
        this.column = value;
    }
    ;
    get pivotColumn() {
        return this.column;
    }
    /**
     * @hidden
     */
    onDoubleClick() {
        this._dblClick = true;
        this.initResizeService();
        this.rowHeaderGroup.grid.autoSizeRowDimension(this.rowHeaderGroup.parent.rootDimension);
    }
    /**
     * @hidden
     */
    initResizeService(event = null) {
        super.initResizeService(event);
        this.colResizingService.rowHeaderGroup = this.rowHeaderGroup;
    }
}
IgxPivotResizeHandleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPivotResizeHandleDirective, deps: [{ token: i0.NgZone }, { token: i0.ElementRef }, { token: i1.IgxPivotColumnResizingService }], target: i0.ɵɵFactoryTarget.Directive });
IgxPivotResizeHandleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxPivotResizeHandleDirective, selector: "[igxPivotResizeHandle]", inputs: { pivotColumn: ["igxPivotResizeHandle", "pivotColumn"], rowHeaderGroup: ["igxPivotResizeHandleHeader", "rowHeaderGroup"] }, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxPivotResizeHandleDirective, decorators: [{
            type: Directive,
            args: [{ selector: '[igxPivotResizeHandle]' }]
        }], ctorParameters: function () { return [{ type: i0.NgZone }, { type: i0.ElementRef }, { type: i1.IgxPivotColumnResizingService }]; }, propDecorators: { pivotColumn: [{
                type: Input,
                args: ['igxPivotResizeHandle']
            }], rowHeaderGroup: [{
                type: Input,
                args: ['igxPivotResizeHandleHeader']
            }] } });
