// igniteui-angular/esm2020/lib/action-strip/grid-actions/grid-actions-base.directive.mjs
import { IgxGridActionButtonComponent } from './grid-action-button.component';
import { Directive, Input, ViewChildren } from '@angular/core';
import { IgxRowDirective } from '../../grids/row.directive';
import * as i0 from "@angular/core";
import * as i1 from "../../icon/icon.service";
export class IgxGridActionsBaseDirective {
    constructor(iconService, differs) {
        this.iconService = iconService;
        this.differs = differs;
        /**
         * Gets/Sets if the action buttons will be rendered as menu items. When in menu, items will be rendered with text label.
         *
         * @example
         * ```html
         *  <igx-grid-pinning-actions [asMenuItems]='true'></igx-grid-pinning-actions>
         *  <igx-grid-editing-actions [asMenuItems]='true'></igx-grid-editing-actions>
         * ```
         */
        this.asMenuItems = false;
    }
    /**
     * @hidden
     * @internal
     */
    get grid() {
        return this.strip.context.grid;
    }
    /**
     * Getter to be used in template
     *
     * @hidden
     * @internal
     */
    get isRowContext() {
        return this.isRow(this.strip.context) && !this.strip.context.inEditMode;
    }
    /**
     * @hidden
     * @internal
     */
    ngAfterViewInit() {
        if (this.asMenuItems) {
            this.buttons.changes.subscribe(() => {
                this.strip.cdr.detectChanges();
            });
        }
    }
    /**
     * Check if the param is a row from a grid
     *
     * @hidden
     * @internal
     * @param context
     */
    isRow(context) {
        return context && context instanceof IgxRowDirective;
    }
}
IgxGridActionsBaseDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridActionsBaseDirective, deps: [{ token: i1.IgxIconService }, { token: i0.IterableDiffers }], target: i0.ɵɵFactoryTarget.Directive });
IgxGridActionsBaseDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.2", type: IgxGridActionsBaseDirective, selector: "[igxGridActionsBase]", inputs: { asMenuItems: "asMenuItems" }, viewQueries: [{ propertyName: "buttons", predicate: IgxGridActionButtonComponent, descendants: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridActionsBaseDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[igxGridActionsBase]'
                }]
        }], ctorParameters: function () { return [{ type: i1.IgxIconService }, { type: i0.IterableDiffers }]; }, propDecorators: { buttons: [{
                type: ViewChildren,
                args: [IgxGridActionButtonComponent]
            }], asMenuItems: [{
                type: Input
            }] } });
