// igniteui-angular/esm2020/lib/calendar/months-view.pipe.mjs
import { Pipe } from '@angular/core';
import { Calendar } from './calendar';
import * as i0 from "@angular/core";
export class IgxMonthViewSlotsCalendar {
    transform(monthViews) {
        return new Array(monthViews);
    }
}
IgxMonthViewSlotsCalendar.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxMonthViewSlotsCalendar, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
IgxMonthViewSlotsCalendar.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxMonthViewSlotsCalendar, name: "IgxMonthViewSlots" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxMonthViewSlotsCalendar, decorators: [{
            type: Pipe,
            args: [{
                    name: 'IgxMonthViewSlots'
                }]
        }] });
export class IgxGetViewDateCalendar {
    constructor() {
        this.calendar = new Calendar();
    }
    transform(index, viewDate, wholeDate = true) {
        const date = this.calendar.timedelta(viewDate, 'month', index);
        return wholeDate ? date : date.getMonth();
    }
}
IgxGetViewDateCalendar.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGetViewDateCalendar, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
IgxGetViewDateCalendar.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGetViewDateCalendar, name: "IgxGetViewDate" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGetViewDateCalendar, decorators: [{
            type: Pipe,
            args: [{
                    name: 'IgxGetViewDate'
                }]
        }], ctorParameters: function () { return []; } });
