// igniteui-angular/esm2020/lib/directives/tooltip/tooltip.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IgxOverlayService } from '../../services/overlay/overlay';
import { IgxTooltipTargetDirective } from './tooltip-target.directive';
import { IgxTooltipComponent } from './tooltip.component';
import { IgxTooltipDirective } from './tooltip.directive';
import * as i0 from "@angular/core";
/**
 * @hidden
 */
export class IgxTooltipModule {
}
IgxTooltipModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTooltipModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxTooltipModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTooltipModule, declarations: [IgxTooltipDirective, IgxTooltipTargetDirective, IgxTooltipComponent], imports: [CommonModule], exports: [IgxTooltipDirective, IgxTooltipTargetDirective] });
IgxTooltipModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTooltipModule, providers: [IgxOverlayService], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxTooltipModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [IgxTooltipDirective, IgxTooltipTargetDirective, IgxTooltipComponent],
                    exports: [IgxTooltipDirective, IgxTooltipTargetDirective],
                    imports: [CommonModule],
                    providers: [IgxOverlayService]
                }]
        }] });
