// igniteui-angular/esm2020/lib/grids/moving/moving.module.mjs
import { NgModule } from '@angular/core';
import { IgxColumnMovingDragDirective } from './moving.drag.directive';
import { IgxColumnMovingDropDirective } from './moving.drop.directive';
import * as i0 from "@angular/core";
export { IgxColumnMovingDragDirective } from './moving.drag.directive';
export { IgxColumnMovingDropDirective } from './moving.drop.directive';
export class IgxColumnMovingModule {
}
IgxColumnMovingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnMovingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxColumnMovingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnMovingModule, declarations: [IgxColumnMovingDropDirective,
        IgxColumnMovingDragDirective], exports: [IgxColumnMovingDropDirective,
        IgxColumnMovingDragDirective] });
IgxColumnMovingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnMovingModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxColumnMovingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxColumnMovingDropDirective,
                        IgxColumnMovingDragDirective
                    ],
                    exports: [
                        IgxColumnMovingDropDirective,
                        IgxColumnMovingDragDirective
                    ]
                }]
        }] });
