// igniteui-angular/esm2020/lib/grids/selection/selection.module.mjs
import { NgModule } from '@angular/core';
import { IgxRowSelectorDirective, IgxHeadSelectorDirective, IgxGroupByRowSelectorDirective } from './row-selectors';
import { IgxGridDragSelectDirective } from './drag-select.directive';
import * as i0 from "@angular/core";
export { IgxRowSelectorDirective, IgxHeadSelectorDirective, IgxGroupByRowSelectorDirective } from './row-selectors';
export { IgxGridDragSelectDirective } from './drag-select.directive';
export class IgxGridSelectionModule {
}
IgxGridSelectionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSelectionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxGridSelectionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSelectionModule, declarations: [IgxRowSelectorDirective,
        IgxGroupByRowSelectorDirective,
        IgxHeadSelectorDirective,
        IgxGridDragSelectDirective], exports: [IgxRowSelectorDirective,
        IgxGroupByRowSelectorDirective,
        IgxHeadSelectorDirective,
        IgxGridDragSelectDirective] });
IgxGridSelectionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSelectionModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxGridSelectionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxRowSelectorDirective,
                        IgxGroupByRowSelectorDirective,
                        IgxHeadSelectorDirective,
                        IgxGridDragSelectDirective
                    ],
                    exports: [
                        IgxRowSelectorDirective,
                        IgxGroupByRowSelectorDirective,
                        IgxHeadSelectorDirective,
                        IgxGridDragSelectDirective
                    ]
                }]
        }] });
