// igniteui-angular/esm2020/lib/grids/hierarchical-grid/hierarchical-grid.module.mjs
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { IgxGridModule } from '../grid/grid.module';
import { IgxChildGridRowComponent, IgxHierarchicalGridComponent } from './hierarchical-grid.component';
import { IgxHierarchicalRowComponent } from './hierarchical-row.component';
import { IgxGridHierarchicalPipe, IgxGridHierarchicalPagingPipe } from './hierarchical-grid.pipes';
import { IgxRowIslandComponent } from './row-island.component';
import { IgxHierarchicalGridCellComponent } from './hierarchical-cell.component';
import * as i0 from "@angular/core";
/**
 * @hidden
 */
export class IgxHierarchicalGridModule {
}
IgxHierarchicalGridModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxHierarchicalGridModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IgxHierarchicalGridModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxHierarchicalGridModule, declarations: [IgxHierarchicalGridComponent,
        IgxHierarchicalRowComponent,
        IgxRowIslandComponent,
        IgxChildGridRowComponent,
        IgxHierarchicalGridCellComponent,
        IgxGridHierarchicalPipe,
        IgxGridHierarchicalPagingPipe], imports: [IgxGridModule], exports: [IgxGridModule,
        IgxHierarchicalGridComponent,
        IgxHierarchicalRowComponent,
        IgxHierarchicalGridCellComponent,
        IgxRowIslandComponent,
        IgxChildGridRowComponent] });
IgxHierarchicalGridModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxHierarchicalGridModule, imports: [[
            IgxGridModule,
        ], IgxGridModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.2", ngImport: i0, type: IgxHierarchicalGridModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IgxHierarchicalGridComponent,
                        IgxHierarchicalRowComponent,
                        IgxRowIslandComponent,
                        IgxChildGridRowComponent,
                        IgxHierarchicalGridCellComponent,
                        IgxGridHierarchicalPipe,
                        IgxGridHierarchicalPagingPipe
                    ],
                    exports: [
                        IgxGridModule,
                        IgxHierarchicalGridComponent,
                        IgxHierarchicalRowComponent,
                        IgxHierarchicalGridCellComponent,
                        IgxRowIslandComponent,
                        IgxChildGridRowComponent
                    ],
                    imports: [
                        IgxGridModule,
                    ],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA]
                }]
        }] });
